src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -340,6 +340,12 @@ public IActionResult GetHistory([FromQuery] WalletHistoryRequest request)
                     var transaction = item.Transaction;
                     var address = item.Address;
 
+                    // We don't show in history transactions that are outputs of staking transactions.
+                    if (transaction.IsCoinStake != null && transaction.IsCoinStake.Value && transaction.SpendingDetails == null)
```

Comment:
I wonder if this might be more readable if you put some of these boolean expressions as properties in FlatHistory. Same with the logic in line 333 above: ``` !t.Address.IsChangeAddress() || (t.Address.IsChangeAddress() && !t.Transaction.IsSpendable())```  Definitely can wait until a future PR after release.

Reply:
Hmm yeah, very good idea.
Ultimately I'd like to build the history in a simple way, like that:
if (this output is a change output) continue;
if (this output is a receive but not staking) process;
if (this output is a receive and staking) continue;

rather than have quite a few conditions that are interlocked. 
Like that It's just way easier to understand what we want to show in the history.